reaktive-testing/src/commonMain/kotlin/com/badoo/reaktive/test/scheduler/TestScheduler.kt

```diff
@@ -53,8 +62,18 @@ class TestScheduler(
         }
 
         fun advanceBy(millis: Long) {
-            timeMillis.addAndGet(millis)
-            listeners.value.forEach { it() }
+            var currentTimerStep = millis
+            while (currentTimerStep > 0) {
+                val advance =
```

Comment:
if `minTimerStep` is between `1 until currentTimerStep` (which starts with `millis` as value) then keep advancing time with the `minTimerStep`, else just advance with the `currentTimerStep`

With every iteration `listeners.value.forEach { it() }` try to `processIfNeeded` to make sure that if we got a new task, it will be executed if needed